internals/ot_process_reminders.py

```diff
@@ -0,0 +1,328 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import requests
+import time
+from datetime import datetime, date
+from typing import Any
+
+from framework import cloud_tasks_helpers
+from framework import origin_trials_client
+from internals.core_models import Stage
+
+RELEASE_DATA_URL = 'https://chromiumdash.appspot.com/fetch_milestone_schedule?mstone='
+
+# Cache of release data
+# Used to avoid multiple fetches from release data API.
+release_cache: dict[str | int, dict[str, str]] = {}
+trials = None
+
+
+def build_trial_data(trial_data: dict[str, Any]) -> dict[str, Any] | None:
+  trial_stage: Stage | None = Stage.query(
+      Stage.origin_trial_id == trial_data['id']).get()
+  if trial_stage is None:
+    logging.exception(f'No stage found for trial {trial_data["id"]}')
+    return None
+  contact_list = trial_stage.ot_emails.copy()
+  contact_list.append(trial_stage.ot_owner_email)
+  contact_list = [s.strip() for s in contact_list]
+
+  # Remove duplicates
+  contact_list = list(set(contact_list))
+
+  trial_info = {
+    'id': trial_data['id'],
+    'name': trial_data['display_name'],
+    'start_milestone': int(trial_data['start_milestone']),
+    'end_milestone': int(trial_data['end_milestone']),
+    'contacts': contact_list
+  }
+  return trial_info
+
+
+# Assemble information about trials that are starting or ending this release.
+def get_trials(release: int) -> dict[str, list[Any]]:
+  trials_list = origin_trials_client.get_trials_list()
+
+  release_trials: dict[str, list[Any]] = {
```

Comment:
Nit: This could be more clear if you define two list variables here and then put them into a dictionary at the return statement of this function.  Or, even return two lists that would be unpacked from the result of each call rather than using a dictionary at all... unless you expect more items in this dict in the future.

Reply:
Agreed! Changed to your latter suggestion. 🙂